### PR TITLE
Pop receipt coherent handling

### LINKED ISSUE #126: Message deletion behaviour is not consistent
## Who's affected

- Any ASQ users that run with a single consumer (with concurrency of 1) and later either scaling out or increasing concurrency level.

## Symptoms

- With a single consumer, message can be deleted after `VisibilityTimeout` elapses. With multiple consumers, this is not going to be possible.

## Analysis

This is an issue with the MSFT ASQ library design. It provided a pop receipt for a message w/o connecting it to visibility timeout. I have [reported the issue](https://github.com/Azure/azure-storage-net/issues/285) and it's now a feature request. 

A possible workaround is to correlate between time when message was received and attempted for deletion [suggested by the ASQ team](https://github.com/Azure/azure-storage-net/issues/285#issuecomment-223887791).

### PR DESCRIPTION
**Pull request by Scooletz**
Fixes #126

### CODE REVIEW

**Comment by Scooletz** [comment]
@SeanFeldman could you review it as you're the author of #126

File: src/Transport/MessageRetrieved.cs [hunk 1]
```diff
@@ -67,4 +82,11 @@ public MessageWrapper Unwrap()
         CloudQueue azureQueue;
         CloudQueueMessage rawMessage;
     }
+
+    class LeaseTimeoutException : Exception
+    {
+        public LeaseTimeoutException() : base("Pop receipt is invalid as it exceeded the next visibile time.")
```
Comment (SeanFeldman) [inline src/Transport/MessageRetrieved.cs]: Should the exception capture the original Visibility time which caused the exception to be raised? Or the message ID that caused this exception to be raised? Thinking as a customer: reading the log that says 

> WARN: Dispatching the message took longer than a visibility timeout. The message will reappear in the queue and will be obtained again. Exception: Pop receipt is invalid as it exceeded the next visibile time.

could be hard to troubleshoot

Comment (SeanFeldman) [inline src/Transport/MessageRetrieved.cs]: "visibile" -> visible

**Comment by SeanFeldman** [comment]
Looks good @Scooletz 
2 minor items and done.

File: src/Transport/MessageRetrieved.cs (hunk 1 above)
Comment (Scooletz) [inline src/Transport/MessageRetrieved.cs]: Cannot agree more @SeanFeldman I'll add the estimated visibility time and the message identifier as well.

**Comment by Scooletz** [comment]
Ping @SeanFeldman

**Comment by SeanFeldman** [comment]
:+1: merged

**Pull request merged**